### PR TITLE
Parse tags into constants

### PR DESCRIPTION
**Pull request by williambaibai**

### CODE REVIEW

File: constants.py [hunk 1]
```diff
@@ -130,3 +132,34 @@ def encode_id():
 
 GYMS = {gym.id: gym for gym in GYMS_LIST}
 
+def parse_tags():
+    result = {}
+    tags = open('tags-grid-view.csv', 'r')
+    tags.readline()
+    line = tags.readline()
+    while line != '':
```
Comment (kq-li) [inline constants.py]: indentation seems screwy here -- 2 space tabs is appdev convention i believe

File: constants.py [hunk 2]
```diff
@@ -130,3 +132,34 @@ def encode_id():
...
+            first = str.find(line, ',')
+            class_name = line[:first]
+            tags_lst = []
+            # tags
+            second = str.find(line, "\",")
+            class_tags = line[first+2:second]
```
Comment (kq-li) [inline constants.py]: `first + 2`

Comment (kq-li) [reply constants.py]: and spaces around the other pluses later

File: constants.py [hunk 3]
```diff
@@ -130,3 +132,34 @@ def encode_id():
...
+                categories = line[third+1:fourth]
+                category_names = str.split(categories, ',')
+            for category_name in category_names:
+                tags_lst.append(category_name)
+            # loop
+            result[class_name] = tags_lst
+            line = tags.readline()
+    return result
+
+TAGS_DICT = parse_tags()
```
Comment (kq-li) [inline constants.py]: add newline at end of file

File: constants.py (hunk 1 above)
Comment (young-k) [reply constants.py]: @kq-li what do you think of just having `while line:` here

**Comment by williambaibai** [comment]
@kq-li ready for review

File: constants.py [hunk 4]
```diff
@@ -130,3 +131,19 @@ def encode_id():
...
 
+def parse_tags():
+    result = {}
+    with open('tags-grid-view.csv', 'r') as tags:
+        reader = csv.reader(tags)
+        next(reader)
+
+        for row in reader:
+            class_name = row[0]
+            tags_lst = []
```
Comment (kq-li) [inline constants.py]: can do `result[class_name] = str.split(row[1], ',') + str.split(row[2], ',')`

File: constants.py [hunk 5]
```diff
@@ -130,3 +131,19 @@ def encode_id():
 
 GYMS = {gym.id: gym for gym in GYMS_LIST}
 
+def parse_tags():
+    result = {}
```
Comment (kq-li) [inline constants.py]: 2 space tabs, might want to adjust that in editor settings

File: constants.py [hunk 6]
```diff
@@ -130,3 +131,19 @@ def encode_id():
...
+        for row in reader:
+            class_name = row[0]
+            tags_lst = []
+            tags_lst.extend(str.split(row[1], ','))
+            tags_lst.extend(str.split(row[2], ','))
+            result[class_name] = tags_lst
+
+    return result
+
+TAGS_DICT = parse_tags()
```
Comment (kq-li) [inline constants.py]: rename to be more descriptive: `TAGS_BY_CLASS_NAME`?

**Comment by williambaibai** [comment]
@kq-li fixed PR issues and added tags to scraper

**Comment by kq-li** [comment]
LGTM, rebase and merge

**Pull request merged**